### ISSUE TITLE
Reorganize subprojects

### ISSUE THREAD

**Issue by acoburn** [trellis-ldp/trellis#105]
There are a lot of top-level directories in the `trellis` repository. It might be nice to add a more logical structure to this, such as:

```
./core/
    api/
    http/
    vocabulary/

./notifications/
    amqp/
    kafka/
    jms/

./components/
    audit/
    agent/
    app/
    io-jena/
    test/
    ...

./platform
    karaf/
    linux/
    distribution/
    osgi/
```

This shouldn't happen until after #66 is merged into master.
